src/link/MachO.zig

```diff
@@ -363,18 +363,30 @@ pub fn openPath(allocator: *Allocator, sub_path: []const u8, options: link.Optio
     self.base.file = file;
 
     // Create dSYM bundle.
-    const d_sym_path = try fmt.allocPrint(allocator, "{s}.dSYM/Contents/Resources/DWARF/", .{sub_path});
-    defer allocator.free(d_sym_path);
-    var d_sym_bundle = try options.emit.?.directory.handle.makeOpenPath(d_sym_path, .{});
-    defer d_sym_bundle.close();
-    const d_sym_file = try d_sym_bundle.createFile(sub_path, .{
-        .truncate = false,
-        .read = true,
-    });
-    self.d_sym = .{
-        .base = self,
-        .file = d_sym_file,
-    };
+    if (options.module) |mod| {
+        const dir = mod.zig_cache_artifact_directory;
+        log.debug("creating {s}.dSYM bundle in {s}", .{ sub_path, dir.path });
+
+        const d_sym_path = try fmt.allocPrint(
+            allocator,
+            "{s}.dSYM" ++ fs.path.sep_str ++ "Contents" ++ fs.path.sep_str ++ "Resources" ++ fs.path.sep_str ++ "DWARF",
```

Comment:
It's not in any special dir within `zig-cache`. It's exactly as you described it. With this patch, it goes directly into `zig-cache/o/<hash>/<bin_name>.dSYM`. For example:

<img width="582" alt="Screenshot 2021-05-14 at 22 15 44" src="https://user-images.githubusercontent.com/1519747/118324163-f41bf600-b501-11eb-8f9b-215488bddd41.png">

Is that what you are describing?

Reply:
looks great! my fault for misreading the code.

Reply:
No probs!

Reply:
The cool thing about dSYM and macOS is that system debugger `lldb` does not require the dSYM bundle to be side-by-side with the executable. It can reside anywhere on the system, and as long as the UUIDs embedded within `LC_UUID` command match, Spotlight service will find and report it to the `lldb`.